lib/Conversion/MooreToCore/MooreToCore.cpp

```diff
@@ -1358,36 +1359,15 @@ struct PowSOpConversion : public OpConversionPattern<PowSOp> {
     Type resultType = typeConverter->convertType(op.getResult().getType());
 
     Location loc = op.getLoc();
-    auto intType = cast<IntType>(op.getRhs().getType());
-    // transform a ** b into scf.for 0 to b step 1 { init *= a }, init = 1
-    Type integerType = rewriter.getIntegerType(intType.getWidth());
-    Value lhsVal = rewriter.create<ConversionOp>(loc, resultType, op.getLhs());
-    Value rhsVal = rewriter.create<ConversionOp>(loc, integerType, op.getRhs());
-    Value constZero = rewriter.create<hw::ConstantOp>(loc, integerType, 0);
-    Value constZeroResult = rewriter.create<hw::ConstantOp>(loc, resultType, 0);
-    Value isNegative = rewriter.create<comb::ICmpOp>(loc, ICmpPredicate::slt,
-                                                     rhsVal, constZero);
-
-    // if the exponent is negative, return 0
-    lhsVal =
-        rewriter.create<comb::MuxOp>(loc, isNegative, constZeroResult, lhsVal);
-    Value upperBound =
-        rewriter.create<comb::MuxOp>(loc, isNegative, constZero, rhsVal);
-
-    Value lowerBound = constZero;
-    Value step = rewriter.create<hw::ConstantOp>(loc, integerType, 1);
-    Value initVal = rewriter.create<hw::ConstantOp>(loc, resultType, 1);
-
-    auto forOp = rewriter.create<scf::ForOp>(
-        loc, lowerBound, upperBound, step, ValueRange(initVal),
-        [&](OpBuilder &builder, Location loc, Value i, ValueRange iterArgs) {
-          auto loopVar = iterArgs.front();
-          auto mul = rewriter.create<comb::MulOp>(loc, lhsVal, loopVar);
-          rewriter.create<scf::YieldOp>(loc, ValueRange(mul));
-        });
-
-    rewriter.replaceOp(op, forOp.getResult(0));
-
+    // utilize MLIR math dialect's math.ipowi to handle the exponentiation of
+    // expression
+    Value result = rewriter.create<mlir::math::IPowIOp>(loc, adaptor.getLhs(),
+                                                        adaptor.getRhs());
+    // cast the result back to MooreToCore i32 type; result is originally
+    // math.ipowi regular i32 so we need to revert it back
+    Value castedResult = rewriter.create<ConversionOp>(loc, resultType, result);
+
+    rewriter.replaceOp(op, castedResult);
```

Comment:
You might be able to do this without the conversion op, in one go (I stole this from the `ShrUOp` case above):
```suggestion
    // utilize MLIR math dialect's math.ipowi to handle the exponentiation of
    // expression
    rewriter.replaceOpWithNewOp<mlir::math::IPowIOp>(
        op, resultType, adaptor.getLhs(), adaptor.getRhs());
```
Not 100% sure this will work, but might be worth a try. Most other ops didn't have to do create a `ConversionOp`, so `math.ipowi` might also be able to get away without :smiley: